README.md

```diff
@@ -1,175 +1,221 @@
- * Circuit: The code that defines what the snark allows.
- * Block: ethereum block
- * Epoch: roll_up block which is a single snark proof of a state transition which contains many transactions.
- * account tree this is the merkle tree that stores a mapping between accounts and balances
- * account_tree_depth (24) this is the number of layers in the account merkle tree
+## Account leaf format
 
-### Tree leaf format
+Each account is represented by a single leaf in the `account_tree`. It is calculated by hashing the following components in the following order:
 
-Each account is represented by a single leaf in the balance tree. It is calculated by hashing the following components in the following order;
 ```python
-leaf = H(pubkey_x, pubkey_y, balance, nonce, token)
+leaf = H(pubkey_x, pubkey_y, balance, nonce, token_type)
 ```
 
-#### Inputs
- * public key X (253 bits)
- * public key Y (253 bits) 
- * token (32 bits)
- * nonce (32 bits)
- * balance (128 bits) 
-![](https://raw.githubusercontent.com/barryWhiteHat/roll_up_token/master/images/database.png)
+*Inputs:*
+
+ * `pubkey_x`: public key X (253 bits)
+ * `pubkey_y`: public key Y (253 bits) 
+ * `balance`: balance (128 bits) 
+ * `nonce`: nonce (32 bits)
+ * `token_type`: token type (32 bits)
+ 
+<p align="center"> 
+<img src ="./images/database.png?raw=true" width=70%>
+</p>
+SNARK storage database
 
-### Deposits
-Each deposit creates a leaf in the smart contract. The smart contract checks that the token type and balance are correct. 
-Anyone can aggregate these deposits into a deposit tree. 
+## Deposit mechanism
+Each deposit creates a leaf in the smart contract. The smart contract checks that the `nonce`, `token_type` and `balance` are correct. Anyone can aggregate these deposits into a `deposit_tree` with a `deposit_root`. 
 
-The operator can add these to the current balance tree by 
+The coordinator can add these to the current balance tree by:
 
-1. Proving that a leaf at the same depth as the deposit tree is zero
-2. Replacing this leaf with the deposit tree 
-3. Using the same merkle proof as the deposit tree to calculate the new root of the balance tree. 
-![](https://raw.githubusercontent.com/barryWhiteHat/roll_up_token/master/images/deposit.pnghttps://raw.githubusercontent.com/barryWhiteHat/roll_up_token/master/images/deposit.png)
+1. Proving that an `empty_node` at the same depth as the `deposit_tree` is empty in the `account_tree`.
+2. Replacing this `empty_node` with the `deposit_root`
+3. Using the same Merkle proof to calculate the new `account_root`.
 
-### Withdraw mechanism
-Leaves can be withdrawn as follows. 
+![](./images/deposit.png?raw=true)
+
+## Withdraw mechanism
+Leaves can be withdrawn on the smart contract as follows. 
 
 The transaction format is 8 bytes:
  
-   * 3 bytes `from`
-   * 3 bytes `to`
-   * 2 bytes `amount`
+   * `from`: 3 bytes 
+   * `to`: 3 bytes 
+   * `nonce`: 2 bytes 
```

Comment:
Why do we need to include the nonce here ?

Reply:
you're right, we don't, the tx index can uniquely identify the withdraw tx